operations/mimir-mixin/dashboards/block-builder.libsonnet

```diff
@@ -76,6 +64,37 @@ local filename = 'mimir-block-builder.json';
         )
       )
     )
+    .addRow(
+      $.row('')
+      .addPanel(
+        $.timeseriesPanel('Partition Lag') +
+        $.panelDescription(
+          'Partition Lag',
+          'Number of records in the backlog of a partition.',
+        ) +
+        $.queryPanel(
+          '(cortex_blockbuilder_scheduler_partition_end_offset{%(job)s} -cortex_blockbuilder_scheduler_partition_committed_offset{%(job)s}) > 0' % { job: $.jobMatcher($._config.job_names.block_builder_scheduler) },
+          '{{partition}}',
+        ) +
+        { fieldConfig+: { defaults+: { custom+: { unit: 'short', fillOpacity: 0 } } } },
+      )
+      .addPanel(
+        $.timeseriesPanel('Pending Jobs by Partition') +
+        $.panelDescription(
+          'Pending jobs by partition',
+          |||
+            Number of pending jobs by a partition.
+
+            Positive number of pending jobs indicates that there is a backlog of work that is waiting to be scheduled to the block-builder.
```

Comment:
Pending = pre-scheduling. They're not eligible yet for scheduling due to our (current) 1-job-per-partition constraint. If you can find a way to state that clearly. :)